### PR TITLE
Make CRDBVersion and Image section optional in crdbcluster 

### PR DESCRIPTION
**Pull request by prafull01**
Currently the image field is mandatory to provide and even if user has provided the `cockroachDBVersion`, he has to provide the image which is not used at all.

This PR makes both the field optional and validates in admission webhooks that user must provide one output to be able to go forward
**Checklist**

* [ ] I have added these changes to the changelog (or it's not applicable).

### CODE REVIEW

File: apis/v1alpha1/cluster_types.go [hunk 1]
```diff
@@ -37,8 +37,8 @@ type CrdbClusterSpec struct {
 	Nodes int32 `json:"nodes"`
 	// Container image information
```
Comment (pseudomuto) [inline apis/v1alpha1/cluster_types.go:38]: Seems like we've got a convention here to prefix these comments with `(Optional) ` for optional fields

File: apis/v1alpha1/webhook.go [hunk 2]
```diff
@@ -83,6 +84,9 @@ func (r *CrdbCluster) Default() {
 func (r *CrdbCluster) ValidateCreate() error {
 	webhookLog.Info("validate create", "name", r.Name)
 
+	if r.Spec.CockroachDBVersion == "" && (r.Spec.Image == nil || r.Spec.Image.Name == "") {
```
Comment (pseudomuto) [inline apis/v1alpha1/webhook.go:87]: Should we do the same check for updates?

Comment (pseudomuto) [reply apis/v1alpha1/webhook.go:87]: Also, I think we're missing a test case for this

File: apis/v1alpha1/cluster_types.go (hunk 1 above)
Comment (prafull01) [reply apis/v1alpha1/cluster_types.go]: Done

File: apis/v1alpha1/webhook.go (hunk 2 above)
Comment (prafull01) [reply apis/v1alpha1/webhook.go:87]: Added for updates as well and added another check if both is provided by the user then also give warning. Also, added the tests for the same

Comment (davidwding) [reply apis/v1alpha1/webhook.go:87]: these checks look a little complex and are duplicated across `ValidateCreate` and `ValidateUpdate` - can you put them into a helper func?

Comment (prafull01) [reply apis/v1alpha1/webhook.go]: Done

File: apis/v1alpha1/webhook_test.go [hunk 3]
```diff
@@ -68,3 +72,62 @@ func TestValidateIngress(t *testing.T) {
...
+		Spec: CrdbClusterSpec{
+			Image: &PodImage{Name: "testImage"},
+			CockroachDBVersion: "v2.1.20",
+		},
+	}
+
+	err := cluster.ValidateUpdate(&oldCluster)
+	require.Error(t, err)
+	require.Equal(t, err.Error(), "you have provided both cockroachDBVersion and cockroach image, please provide only one")
+}
```
Comment (pseudomuto) [inline apis/v1alpha1/webhook_test.go:133]: nit: missing new line

**Review by pseudomuto — APPROVED** [review]
LGTM 👍 

I think we could make those tests table tests to avoid some duplication, but no biggie.

File: apis/v1alpha1/webhook_test.go (hunk 3 above)
Comment (prafull01) [reply apis/v1alpha1/webhook_test.go]: Done, added table driven tests as well

**Pull request merged**